packages/libs/types/src/Predicate.ts

```diff
@@ -0,0 +1,7 @@
+/**
+ * The most common predicate values are 'keys-all', 'keys-2' and 'keys-any'.
+ * However, it is allowed to use any string. If a predicate is undefined,
+ * the default value used by Pact is 'keys-all'.
+ * @alpha
+ */
+export type TPredicate = string | undefined;
```

Comment:
I think we also need to expose a type like `CommonPredicate`.
and then in the client use it instead of `TPredicate ` to avoid breaking type suggestion.
I will add one more overload to the functions in the client to allow normal string.
```suggestion
export type TPredicate = string | undefined;
export type CommonPredicate = "keys-all" | "keys-2" | "keys-any"
```